apps/networkmonitor/networkmonitor.nim

```diff
@@ -65,15 +72,25 @@ proc setConnectedPeersMetrics(discoveredNodes: seq[Node],
       warn "could not get secp256k1 key", typedRecord=typedRecord.get()
       continue
 
+    let peerRes = toRemotePeerInfo(discNode.record)
+
+    let peerInfo = peerRes.valueOr():
+      warn "error converting record to remote peer info", record=discNode.record
+      continue
+
     # create new entry if new peerId found
-    let peerId = secp256k1.get().toHex()
+    let peerId = $peerInfo.peerId
     let customPeerInfo = CustomPeerInfo(peerId: peerId)
     if not allPeers.hasKey(peerId):
       allPeers[peerId] = customPeerInfo
+      newPeers += 1
+    else:
+      info "already seen", peerId=peerId
 
     allPeers[peerId].lastTimeDiscovered = currentTime
     allPeers[peerId].enr = discNode.record.toURI()
     allPeers[peerId].enrCapabilities = discNode.record.getCapabilities().mapIt($it)
+    allPeers[peerId].discovered += 1
```

Comment:
Perhaps time to assign `allPeers[peerId]` here to a variable? Just to improve readability rather than repeated `allPeers[peerId]` throughout the rest of the proc.